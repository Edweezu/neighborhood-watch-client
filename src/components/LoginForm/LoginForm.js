import React from 'react'
import { Link } from 'react-router-dom'
import Footer from '../../components/Footer/Footer'

class LoginForm extends React.Component {

    state = {
        error: null,
        password: 'Testing123!',
        username: 'demo'
    }
    
    render () {
        // const { error, username, password } = this.state
        const { error } = this.state

        return (
            <section>
            <form className='LoginForm' onSubmit={this.handleSubmit}>
            <div role='alert'>
                {error && <p className='red'>{error}</p>}
            </div>
            <div className='LoginForm__signupElement'>
                <div className='LoginForm__signupLabel'>
                    <label htmlFor='username'>
                        Username    
                    </label>
                    <span className='astrik'>
                        *
                    </span>
                
                
                    {/* <input id='username' name='username' type='text' placeholder='demo' value={username} onChange={this.handleChangeUser} required/> */}
                    <input id='username' name='username' type='text' placeholder='demo' required/> 
                </div> 
            </div>
            <div className='LoginForm__signupElement'>
                <div className='LoginForm__signupLabel'>
                    <label htmlFor='password'>
                        Password    
                    </label>
                    <span className='astrik'>*</span>
                     
                    {/* <input id='password' name='password' type='password' placeholder='Testing123!' value={password} onChange={this.handleChangePassword} required/> */}
                    <input id='password' name='password' type='password' placeholder='Testing123!'  required/> 
                </div>
            </div>
            {/* <div className='signin-button'>
                <button className="btn btn-sign-in" type='submit'>
                    Sign In 
                </button>
            </div> */}
            <Link to='/category/1'>Sign In</Link>
            <div className='LoginForm__loginDemo'>
                <h4>Demo Account</h4>
                <p>Username: demo</p>
                <p>Password: Testing123!</p>
            </div>
            <div className='LoginForm__redirect'>
                <Link to='/register'>Don't have an account? Create one.</Link>
            </div>
        </form>
        <Footer />
        </section>
        )
    }
}

export default LoginForm