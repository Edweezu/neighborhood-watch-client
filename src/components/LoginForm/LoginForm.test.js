import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import LoginForm from './LoginForm'


describe('LoginForm component', () => {
    const props = {
        onLoginSuccess: () => {

        },
    }

    it('renders the given component', () => {
        const wrapper = shallow(<LoginForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it(`renders the component's given props`, () => {
        const wrapper = shallow(<LoginForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})