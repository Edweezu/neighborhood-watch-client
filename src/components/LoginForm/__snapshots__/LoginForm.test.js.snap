// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the component's given props 1`] = `
<section>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <span
          className="astrik"
        >
          *
        </span>
        <input
          id="username"
          name="username"
          placeholder="demo"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <span
          className="astrik"
        >
          *
        </span>
        <input
          id="password"
          name="password"
          placeholder="Testing123!"
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="signin-button"
    >
      <button
        className="btn btn-sign-in"
        type="submit"
      >
        Sign In
      </button>
    </div>
    <div
      className="LoginForm__loginDemo"
    >
      <h4>
        Demo Account
      </h4>
      <p>
        Username: demo
      </p>
      <p>
        Password: Testing123!
      </p>
    </div>
    <div
      className="LoginForm__redirect"
    >
      <Link
        to="/register"
      >
        Don't have an account? Create one.
      </Link>
    </div>
  </form>
  <Footer />
</section>
`;

exports[`LoginForm component renders the given component 1`] = `
<section>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <div
      role="alert"
    />
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="username"
        >
          Username
        </label>
        <span
          className="astrik"
        >
          *
        </span>
        <input
          id="username"
          name="username"
          placeholder="demo"
          required={true}
          type="text"
        />
      </div>
    </div>
    <div
      className="LoginForm__signupElement"
    >
      <div
        className="LoginForm__signupLabel"
      >
        <label
          htmlFor="password"
        >
          Password
        </label>
        <span
          className="astrik"
        >
          *
        </span>
        <input
          id="password"
          name="password"
          placeholder="Testing123!"
          required={true}
          type="password"
        />
      </div>
    </div>
    <div
      className="signin-button"
    >
      <button
        className="btn btn-sign-in"
        type="submit"
      >
        Sign In
      </button>
    </div>
    <div
      className="LoginForm__loginDemo"
    >
      <h4>
        Demo Account
      </h4>
      <p>
        Username: demo
      </p>
      <p>
        Password: Testing123!
      </p>
    </div>
    <div
      className="LoginForm__redirect"
    >
      <Link
        to="/register"
      >
        Don't have an account? Create one.
      </Link>
    </div>
  </form>
  <Footer />
</section>
`;
